src/test/unit/algorithms/ConnectionsTest.cpp

```diff
@@ -481,6 +482,37 @@ namespace nupic {
     NTA_CHECK(ret == 0) << "Failed to delete " << filename;
   }
 
+  void ConnectionsTest::testSaveLoad()
+  {
+    const char* filename = "ConnectionsSerialization.tmp";
+    Connections c1(1024, 1024, 1024), c2;
+    setupSampleConnections(c1);
+
+    Segment segment;
+    Cell cell, presynapticCell;
+
+    cell.idx = 10;
+    presynapticCell.idx = 400;
+    segment = c1.createSegment(cell);
+    c1.createSynapse(segment, presynapticCell, 0.5);
+    c1.destroySegment(segment);
+
+    computeSampleActivity(c1);
+
+    ofstream os(filename, ios::binary);
```

Comment:
Can you use a stringstream instead? Example:
https://github.com/numenta/nupic.core/blob/42d66a13dffc2647ea852ed4d2632f751b66ecf4/src/test/unit/algorithms/FastCLAClassifierTest.cpp#L124-L126